aws_xray_sdk/core/serverless_lambda_context.py

```diff
@@ -0,0 +1,130 @@
+import os
+import logging
+
+from .models.facade_segment import FacadeSegment
+from .models.segment import Segment
+from .models.mimic_segment import MimicSegment
+from .context import CXT_MISSING_STRATEGY_KEY
+from .lambda_launcher import LambdaContext
+
+
+log = logging.getLogger(__name__)
+
+
+class ServerlessLambdaContext(LambdaContext):
```

Comment:
I'm a little unclear on why this additional Context is needed. It looks like `LambdaContext` does a no-op and logs a warning when working with `segments`. Since this is an additive change, it seems that the two Context classes can be merged into one. Is there a backwards compatibility issue this is solving?

Reply:
That is a good point that you make. I did consider just adding the existing features to the LambdaContext, but I didn't want to confuse customers and suddenly allow them to create "segments" in Lambda apart from using it in a serverless application.

Reply:
Could that just be documented though, that calling put_segment in a Lambda context ends up  creating subsegments? That's what the ServerlessLambdaContext is doing. Mostly I'm not sure why there needs to be a distinction between ServerlessLambda and Lambda, and I think that was what @haotianw465 was hinting at in an earlier comment.

I don't think we have to worry about the case wherein we would want a segment in Lambda to be represented differently than a local subsegment, but correct me if I'm wrong!

Reply:
Hmmm in that case, the usage requirements would be different; What happens if a customer attempts to create a subsegment when no parent segment (mimic segment) has been created (even though the facade segment exists)? Do we automatically attach that to the facade segment or do we not permit it? What if the customer runs put_segment then end_subsegment? I don't know if there are right or wrong answers here, but I think it requires a larger discussion.

Wouldn't we want a behavior to be consistent throughout the SDKs? From my knowledge, I don't think that the other SDKs support begin_segment explicitly in Lambda. This would mean that in Python, you would be able to begin segments and end segments as expected, but the rules will be different for other SDKs.

Reply:
Decision has been made to discuss this issue sometime in a future date. For right now, we'll keep it simple and have the middlewares only begin subsegments when they usually begin segments.